src/transformers/models/colpali/convert_colpali_weights_to_hf.py

```diff
@@ -0,0 +1,481 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Convert ColPali weights."""
+
+import argparse
+import glob
+from pathlib import Path
+from typing import Any, Dict, cast
+
+import torch
+from huggingface_hub import snapshot_download
+from PIL import Image
+from safetensors import safe_open
+
+from transformers.models.colpali import ColPaliForRetrieval, ColPaliProcessor
+from transformers.models.colpali.configuration_colpali import ColPaliConfig
+from transformers.utils import logging
+
+
+logging.set_verbosity_info()
+logger = logging.get_logger(__name__)
+
+
+ORIGINAL_DTYPE = torch.bfloat16
+TOLERANCE = 1e-3
+
+
+# Copied from https://huggingface.co/vidore/colpali-v1.2-merged/blob/main/config.json
+ORIGINAL_CONFIG: Dict[str, Any] = {
```

Comment:
no no, if this is from the hub just use AutoConfig.from_pretrained, with a revisioon!